test/unit/mesh-gateway-deployment.bats

```diff
@@ -654,3 +654,30 @@ key2: value2' \
   actual=$(echo $ca_cert_volume | jq -r '.secret.items[0].key' | tee /dev/stderr)
   [ "${actual}" = "key" ]
 }
+
+#--------------------------------------------------------------------
+# global.federation.enabled
+
+@test "meshGateway/Deployment: -expose-servers set when federation.enabled=true" {
+  cd `chart_dir`
+  local actual=$(helm template \
+      -x templates/mesh-gateway-deployment.yaml  \
+      --set 'meshGateway.enabled=true' \
+      --set 'connectInject.enabled=true' \
+      --set 'global.federation.enabled=true' \
+      . | tee /dev/stderr |
+      yq '.spec.template.spec.containers[0].command | join(" ") | contains("-expose-servers")' | tee /dev/stderr)
+  [ "${actual}" = "true" ]
+}
+
+@test "meshGateway/Deployment: -expose-servers not set when federation.enabled=false" {
+  cd `chart_dir`
+  local actual=$(helm template \
+      -x templates/mesh-gateway-deployment.yaml  \
+      --set 'meshGateway.enabled=true' \
+      --set 'connectInject.enabled=true' \
```

Comment:
I don't think we need`connectInject.enabled` set to `true` here and in the above test.

Reply:
Mesh gateways don't render when connect inject is not enabled. There's a `fail` at the top of the file.